### PR TITLE
PSP-9312 | Document type purpose / definition

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx [hunk 1]
```diff
@@ -55,6 +55,14 @@ export const SelectedDocumentHeader: React.FunctionComponent<ISelectedDocumentHe
     [documentTypes],
   );
 
+  const selectedDocumentType = useMemo(() => {
+    const documentType = documentTypes.find(d => Number(document.documentTypeId) === d.id);
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx:59]: unrelated to these changes, but isn't the documentTypeId a number? at least I think it is on the backend.

File: source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx [hunk 2]
```diff
@@ -55,6 +55,14 @@ export const SelectedDocumentHeader: React.FunctionComponent<ISelectedDocumentHe
     [documentTypes],
   );
 
+  const selectedDocumentType = useMemo(() => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx:58]: nit: This variable appears to contain the purpose and not the document type

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+    document_types = []
+
+    # Get the document Type
+    for i in range(2, doctype_sheet.nrows):
+        doc_type_code = doctype_sheet.cell_value(i, 1)
+        doc_type_name = doctype_sheet.cell_value(i, 2)
+        doc_type_purpose = doctype_sheet.cell_value(i, 3)
+        # display_order = int(doctype_sheet.cell_value(i, 3))
+        display_order = 8
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:84]: ?

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    metadata_file_location = ("xml_definitions/metadata_definitions.xlsx")
+
+    # To open Workbook
+    wb = xlrd.open_workbook(metadata_file_location)
+    metadata_sheet = wb.sheet_by_index(0)
+
+    metadata_types = []
+
+    # Get the document Type
+    for i in range(3, metadata_sheet.nrows):
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:138]: nit: magic number(3)

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    wb = xlrd.open_workbook(loc_match)
+    sheet = wb.sheet_by_index(0)
+
+    doctype_sort_order = []
+
+    for i in range(3, sheet.nrows):
+        print("--------------" + str(i))
+        current_row_name = sheet.cell_value(i, 1)
+
+        if current_row_name != "":
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:163]: why is this necessary?

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        return {"document_types": [value.toJson() for value in self.document_types],
+                "metadata_types": [value.toJson() for value in self.metadata_types],
+                "remove_lingering_document_types": self.remove_lingering_document_types,
+                "remove_lingering_metadata_types": self.remove_lingering_metadata_types
+                }
+
+
+def load_document_definitions():
+    # Give the location of the file
+    doctype_location = ("xml_definitions/doctype_definitions.xlsx")
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:70]: nit: perhaps a common function to get a file, open the workbook, extract the sheet, and possibly index to the first valid row? that would cut down on some of this duplicated code.

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    doctype_sort_order = []
+
+    for i in range(3, sheet.nrows):
+        print("--------------" + str(i))
+        current_row_name = sheet.cell_value(i, 1)
+
+        if current_row_name != "":
+            doc_type_name = current_row_name
+
+        if doc_type_name not in doctype_sort_order:
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:166]: Hmm, we may want to throw here instead, I feel like I'd rather fail fast then accidentally generate a json with missing data.

File: tools/mayan_sync_helper/mayan_sync_transformer.py [hunk 8]
```diff
@@ -0,0 +1,220 @@
+
```
Comment (devinleighsmith) [inline tools/mayan_sync_helper/mayan_sync_transformer.py:1]: please either add some doc for this, or update the make file.

Comment (devinleighsmith) [reply tools/mayan_sync_helper/mayan_sync_transformer.py:1]: Thanks for adding this!

**Review by devinleighsmith — APPROVED** [review]

File: source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx (hunk 1 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx:59]: it is, the problem is that on the form is a string.

File: tools/mayan_sync_helper/mayan_sync_transformer.py (hunk 5 above)
Comment (FuriousLlama) [reply tools/mayan_sync_helper/mayan_sync_transformer.py:163]: Sometimes the document definition takes multiple rows, leaving the row name empty

File: tools/mayan_sync_helper/mayan_sync_transformer.py (hunk 8 above)
Comment (FuriousLlama) [reply tools/mayan_sync_helper/mayan_sync_transformer.py:1]: good idea. I will add a new pr with a readme
